src/lib.rs

```diff
@@ -1,3 +1,95 @@
+//! A reasonably simple entity component system (ECS),
+//! with a design based on [hecs](https://github.com/Ralith/hecs).
+//!
+//! Compared to hecs, but it drops all optimisations which require additional API surface.
+//! Furthermore, it is _not_ thread-safe, uses prepared queries only,
+//! and adds a type map used to hold resources.
+//!
+//! # Examples
+//!
+//! A simple example that creates two entities with different components,
+//! and iterates over them using queries on different combinations of components:
+//!
+//! ```
+//! # use rs_ecs::*;
+//!
+//! let mut world = World::new();
+//!
+//! let entity = world.alloc();
+//! world.insert(entity, (42_u32, true));
+//!
+//! let entity = world.alloc();
+//! world.insert(entity, (23_u32, "hello".to_string()));
+//!
+//! for number in Query::<&u32>::new().iter(&world) {
+//!     println!("{}", number);
+//! }
+//!
+//! for (_entity, number, string) in Query::<(&Entity, &u32, &String)>::new().iter(&world) {
+//!     println!("{}, {}", string, number);
+//! }
+//! ```
+//!
+//! ----
+//!
+//! A more complex example using custom components, resources, and functions as systems:
+//!
+//! ```
+//! use rs_ecs::*;
+//!
+//! // Components
+//! struct Position {
+//!     pub x: f32,
+//!     pub y: f32,
+//! }
+//!
+//! struct Velocity {
+//!     pub x: f32,
+//!     pub y: f32,
+//! }
+//!
+//! // A resource for the length of the time step
+//! struct Delta {
```

Comment:
I think this would be a good tuple struct.

Reply:
(The `dt` will still be visible in the let binding.)

Reply:
Removed the long example.